src/main/kotlin/it/pagopa/wallet/services/WalletService.kt

```diff
@@ -172,26 +179,34 @@ class WalletService(
                             )
                     )
                     .map { hostedOrderResponse ->
-                        Triple(hostedOrderResponse, wallet, Pair(orderId, contractId))
+                        val newDetails =
+                            when (sessionInputDataDto) {
+                                is SessionInputCardDataDto -> wallet.details
+                                is SessionInputPayPalDataDto ->
+                                    PayPalDetails(null, sessionInputDataDto.pspId)
+                                else ->
+                                    throw InternalServerErrorException("Unhandled session input")
+                            }
+                        val updatedWallet =
+                            wallet.copy(
+                                contractId = ContractId(contractId),
+                                status = WalletStatusDto.INITIALIZED,
```

Comment:
Just as reminder, `INITIALIZED` status here is right only for cards onboardings, for apm ones it must be updated to `VALIDATION_REQUESTED`